src/main/java/sg/dex/starfish/impl/file/FileAsset.java

```diff
@@ -53,22 +55,13 @@ public static FileAsset create(File f) {
      * @return FileAsset instance created using given params
      */
     public static FileAsset create(File f, Map <String,Object> metaData) {
-        validateFileExist(f);
-        return new FileAsset(buildMetadata( metaData), f);
-    }
-
-    /**
-     * Method to validate if the file exist.
-     * @param f file to be validated
-     */
-    private static void validateFileExist(File f){
-
         if(f== null || !f.exists()){
```

Comment:
I don't think this is the right place to check for file existence. Shouldn't this check happen when we try to access the file?

Logic should be the same as for a remote asset.